File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.py
class Solution(object):
    def eraseOverlapIntervals(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: int
        """
        res = 0
        curEnd = float("-inf")

        for i in sorted(intervals, key=lambda x: x[1]):
            if i[0] >= curEnd:
                curEnd = i[1]
            else:
                res += 1

        return res
        

File: 0435-non-overlapping-intervals/0435-non-overlapping-intervals.go
func eraseOverlapIntervals(intervals [][]int) int {
    slices.SortFunc(intervals, func(a, b []int) int {
        return a[1] - b[1]
    })

    cur := intervals[0][1]
    cnt := 1
    for i := 1; i < len(intervals); i++ {
        if intervals[i][0] >= cur {
            cnt += 1
            cur = intervals[i][1]
        }
    }

    return len(intervals) - cnt
}